pkg/datagatherer/k8s/generic.go

```diff
@@ -110,3 +121,16 @@ func namespaceResourceInterface(iface dynamic.NamespaceableResourceInterface, na
 	}
 	return iface.Namespace(namespace)
 }
+
+// generateFieldSelector creates a field selector string from a list of
+// namespaces to exclude.
+func generateFieldSelector(excludeNamespaces []string) string {
```

Comment:
Since we could use `FieldSelectors` for other type of filtering, I'd rename it to `excludeNamespacesToFieldSelector` or something that makes it explicit it is for excluding namespaces.

Reply:
I imagined that if we added more things to the field selector they would also be handled by this function. Then we could split stuff out later to other functions if it gets too big.